File: calorie_calc.py
"""
حساب BMR و TDEE بمعادلة Mifflin-St Jeor — Server-Side حصرًا.
حدود السلامة (الحد الأدنى الآمن، أقصى عجز/فائض) مصدرها nutrition_ai/safety.py حصرًا —
هذا الملف لا يقرر أي رقم حرج بنفسه.
"""
from nutrition_ai import safety

ACTIVITY_FACTORS = {
    "sedentary": 1.2,
    "light": 1.375,
    "moderate": 1.55,
    "very_active": 1.725,
}

ACTIVITY_LABELS = {
    "sedentary": "خامل — بدون تمرين",
    "light": "نشاط خفيف — 1-3 أيام بالأسبوع",
    "moderate": "نشاط متوسط — 4-5 أيام بالأسبوع",
    "very_active": "نشاط عالي جدًا — 6-7 أيام بالأسبوع",
}


def calculate(age: int, weight_kg: float, height_cm: float, sex: str,
              goal: str, activity_level: str) -> dict:
    if sex == "male":
        bmr = 10 * weight_kg + 6.25 * height_cm - 5 * age + 5
    else:
        bmr = 10 * weight_kg + 6.25 * height_cm - 5 * age - 161

    factor = ACTIVITY_FACTORS.get(activity_level, 1.2)
    tdee = bmr * factor

    if goal == "lose":
        target = tdee - 500
    elif goal == "gain":
        target = tdee + 400
    else:
        target = tdee

    safe_target, safety_warning = safety.enforce_safe_target(target, tdee, sex, goal)

    water_target_ml = round(weight_kg * 33)  # تقدير شائع ومعقول، قابل للتعديل لاحقًا

    return {
        "bmr": round(bmr),
        "tdee": round(tdee),
        "calorie_target": safe_target,
        "water_target_ml": water_target_ml,
        "safety_warning": safety_warning,
    }

File: shared/nutrition-engine/calorieCalc.ts
/**
 * منفذ حرفي من calorie_calc.py — حساب BMR وTDEE بمعادلة Mifflin-St Jeor. حدود السلامة مصدرها
 * safety.ts حصرًا — هذا الملف لا يقرر أي رقم حرج بنفسه.
 */
import * as safety from "./safety.js";
import { pyRound } from "./pyRound.js";

export const ACTIVITY_FACTORS: Record<string, number> = {
  sedentary: 1.2,
  light: 1.375,
  moderate: 1.55,
  very_active: 1.725,
};

export const ACTIVITY_LABELS: Record<string, string> = {
  sedentary: "خامل — بدون تمرين",
  light: "نشاط خفيف — 1-3 أيام بالأسبوع",
  moderate: "نشاط متوسط — 4-5 أيام بالأسبوع",
  very_active: "نشاط عالي جدًا — 6-7 أيام بالأسبوع",
};

export interface CalorieCalcResult {
  bmr: number;
  tdee: number;
  calorie_target: number;
  water_target_ml: number;
  safety_warning: string | null;
}

export function calculate(
  age: number, weightKg: number, heightCm: number, sex: string,
  goal: string, activityLevel: string,
): CalorieCalcResult {
  const bmr = sex === "male"
    ? 10 * weightKg + 6.25 * heightCm - 5 * age + 5
    : 10 * weightKg + 6.25 * heightCm - 5 * age - 161;

  const factor = ACTIVITY_FACTORS[activityLevel] ?? 1.2;
  const tdee = bmr * factor;

  let target: number;
  if (goal === "lose") target = tdee - 500;
  else if (goal === "gain") target = tdee + 400;
  else target = tdee;

  const { target: safeTarget, warning: safetyWarning } = safety.enforceSafeTarget(target, tdee, sex, goal);

  const waterTargetMl = pyRound(weightKg * 33); // تقدير شائع ومعقول، قابل للتعديل لاحقًا

  return {
    bmr: pyRound(bmr),
    tdee: pyRound(tdee),
    calorie_target: safeTarget,
    water_target_ml: waterTargetMl,
    safety_warning: safetyWarning,
  };
}
